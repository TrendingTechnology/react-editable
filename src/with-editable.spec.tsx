import withEditable from "./with-editable";

import * as React from "react";
import { mount } from "enzyme";
import MockComponent from "./__mocks__/mock-component";

const MockComponentWithEditable = withEditable(MockComponent);

describe("withEditable", () => {
  test("should create the correct displayName", () => {
    expect(MockComponentWithEditable.displayName).toBe(
      "WithEditable(MockComponent)"
    );

    let ComponentWithName: React.ComponentType<any> = () => null;
    expect(withEditable(ComponentWithName).displayName).toBe(
      "WithEditable(ComponentWithName)"
    );

    expect(withEditable(() => null).displayName).toBe(
      "WithEditable(Component)"
    );
  });

  test("shouldn't fatal", () => {
    expect(() => (
      <MockComponentWithEditable value={undefined} testProp={1} />
    )).not.toThrow();
  });

  test("should hoist MockComponent props", () => {
    expect(MockComponentWithEditable.propTypes).toHaveProperty("testProp");
  });

  test("should match snapshot", () => {
    expect(
      mount(<MockComponentWithEditable value={undefined} testProp={1} />)
    ).toMatchSnapshot();
  });
});
