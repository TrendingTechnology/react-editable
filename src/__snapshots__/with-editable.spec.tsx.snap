// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withEditable should match snapshot 1`] = `
<WithEditable(MockComponent)
  testProp={1}
>
  <Editable>
    <MockComponent
      onCancel={[Function]}
      onChange={[Function]}
      onCommit={[Function]}
      onStart={[Function]}
      status="PRESENTING"
      testProp={1}
    />
  </Editable>
</WithEditable(MockComponent)>
`;
